ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/__init__.py

```diff
@@ -0,0 +1,346 @@
+        '''
+        if template_path := get(self._hostvars, "switch.ip_addressing.mlag_ibgp_peering_ip_secondary"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['vrf'] = {
+                'mlag_ibgp_peering_ipv4_pool': mlag_ibgp_peering_ipv4_pool
+            }
+            return self.template_var(template_path, template_vars)
+
+        offset = self._mlag_primary_id - 1
+        return self._ip(mlag_ibgp_peering_ipv4_pool, 31, offset, 1)
+
+    def mlag_ip_primary(self) -> str:
+        '''
+        Return IP for MLAG Primary
+
+        Default pool is "switch.mlag_peer_ipv4_pool"
+        Default offset from pool is `(mlag_primary_id - 1) * 2`
+        '''
+        if template_path := get(self._hostvars, "switch.ip_addressing.mlag_ip_primary"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['mlag_primary_id'] = self._mlag_primary_id
+            template_vars['mlag_secondary_id'] = self._mlag_secondary_id
+            template_vars['switch_data'] = {
+                'combined': {
+                    'mlag_peer_ipv4_pool': self._mlag_peer_ipv4_pool
+                }
+            }
+            return self.template_var(template_path, template_vars)
+
+        offset = self._mlag_primary_id - 1
+        return self._ip(self._mlag_peer_ipv4_pool, 31, offset, 0)
+
+    def mlag_ip_secondary(self) -> str:
+        '''
+        Return IP for MLAG Secondary
+
+        Default pool is "switch.mlag_peer_ipv4_pool"
+        Default offset from pool is `((mlag_primary_id - 1) * 2) + 1`
+        '''
+        if template_path := get(self._hostvars, "switch.ip_addressing.mlag_ip_secondary"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['mlag_primary_id'] = self._mlag_primary_id
+            template_vars['mlag_secondary_id'] = self._mlag_secondary_id
+            template_vars['switch_data'] = {
+                'combined': {
+                    'mlag_peer_ipv4_pool': self._mlag_peer_ipv4_pool
+                }
+            }
+            return self.template_var(template_path, template_vars)
+
+        offset = self._mlag_primary_id - 1
+        return self._ip(self._mlag_peer_ipv4_pool, 31, offset, 1)
+
+    def mlag_l3_ip_primary(self) -> str:
+        '''
+        Return IP for L3 Peerings for MLAG Primary
+
+        Default pool is "switch.mlag_peer_l3_ipv4_pool"
+        Default offset from pool is `(mlag_primary_id - 1) * 2`
+        '''
+        if template_path := get(self._hostvars, "switch.ip_addressing.mlag_l3_ip_primary"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['mlag_primary_id'] = self._mlag_primary_id
+            template_vars['mlag_secondary_id'] = self._mlag_secondary_id
+            template_vars['switch_data'] = {
+                'combined': {
+                    'mlag_peer_l3_ipv4_pool': self._mlag_peer_l3_ipv4_pool
+                }
+            }
```

Comment:
Wondering if somehow we could manage to abstract this "repetitive" block in some function ?

looks like in most cases we need to
1. check what is the template
2. add some template_vars which we could list
3. the combined/ vrf part is a bit mort tricky as I guess this is to not deprecate the existing templates?

Reply:
I have implemented a small helper function, but it is very limited how much we can save, since the template vars are slightly different for most of the functions. Let me know what you think